src/components/Toolbar.js

```diff
@@ -1,98 +1,158 @@
+    setInitialScrollLeft(scrollLeft)
+    setLastScrollLeft(scrollLeft)
+  }, [])
+
+  useEffect(() => {
+    const window90 = window.innerWidth * 0.9
+    const toolbarElement = document.getElementById('toolbar')
+    const scrollLeft = toolbarElement.scrollLeft
+    const scrollWidth = toolbarElement.scrollWidth
+    setToolbarElementScrollWidth(scrollWidth)
+
+    /** set event listeners start */
+    window.addEventListener('mouseup', clearHoldTimer)
+    window.addEventListener('touchend', clearHoldTimer)
+    window.addEventListener('resize', () => {
+      if (toolbarElementScrollWidth > window90) setLeftArrowElementClassName('')
+      else setLeftArrowElementClassName('hidden')
+    })
+    /** set event listeners end */
+
+    if (toolbarElementScrollWidth > window90) setLeftArrowElementClassName('')
+    else setLeftArrowElementClassName('hidden')
+
+    if (scrollLeft < initialScrollLeft) setRightArrowElementClassName('')
+    else if (scrollLeft >= initialScrollLeft) setRightArrowElementClassName('hidden')
+
+    if (scrollLeft === 0) setLeftArrowElementClassName('hidden')
+    else setLeftArrowElementClassName('')
+
+  })
+
   return (
-    'ontouchstart' in window ||
-    navigator.maxTouchPoints > 0 ||
-    navigator.msMaxTouchPoints > 0
-  )
-}
-
-const clearHoldTimer = () => {
-  overlayHide()
-  clearTimeout(holdTimer)
-}
-
-(() => {
-  window.addEventListener('mouseup', clearHoldTimer)
-  window.addEventListener('touchend', clearHoldTimer)
-})()
-
-const shortcutIds = [
-  'outdent',
-  'indent',
-  'delete',
-  'toggleContextView',
-  'exportContext',
-  'search',
-]
-
-let holdTimer
-
-const onHoldDownShortcut = id => {
-  // on chrome setTimeout doesn't seem to work on the first click, clearing it before hand fixes the problem
-  clearTimeout(holdTimer)
-  holdTimer = setTimeout(() => {
-    overlayReveal(id)
-  }, SHORTCUT_HINT_OVERLAY_TIMEOUT)
-}
-
-export const Toolbar = connect(({ toolbarOverlay, settings: { dark } }) => ({
-  dark,
-  toolbarOverlay
-}))(
-  ({
-    dark,
-    toolbarOverlay
-  }) => (
-    <div>
-      <div className={'toolbar-container'}>
-        <div className={'toolbar'}>
-          {shortcutIds.map(id => {
-            const { name, svg: Icon, exec } = shortcutById(id)
-            return (
-              <div
-                key={name}
-                id={id}
-                className='toolbar-icon'
-                onMouseDown={() => onHoldDownShortcut(id)}
-                onMouseOver={() => {
-                    if (toolbarOverlay) overlayReveal(id)
+      <div>
+        <div className='toolbar-container'>
+          <div
+            id='toolbar'
+            className='toolbar'
+            onScroll={e => {
+              const target = e.target
+              const scrollLeft = target.scrollLeft
+              const scrollDifference = lastScrollLeft - scrollLeft
+
+              if (scrollDifference >= 5 || scrollDifference <= -5) scrollPrioritize(true)
```

Comment:
Why create another state variable instead of just calling `overlayHide`?

Let's try to accomplish this without adding more complexity to state.

Reply:
Got it.